### PR TITLE
feat(import_users): support user's password in plain text

### PR DESCRIPTION
**Pull request by HJianBo**
Fixes https://emqx.atlassian.net/browse/EMQX-11768

Release version: v/e5.5.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- ~Added property-based tests for code which performs user input validation~
- ~Changed lines covered in coverage report~
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [x] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- ~Schema changes are backward compatible~

## Checklist for CI (.github/workflows) changes

- ~If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
- [x] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl [hunk 1]
```diff
@@ -482,10 +445,143 @@ group_match_spec(UserGroup, QString) ->
...
+%% Example: data/user-credentials.json
+reader(json, Data, Convertor) when is_binary(Data) ->
+    case emqx_utils_json:safe_decode(Data, [return_maps]) of
+        {ok, List} ->
+            reader(prepared_user_list, List, Convertor);
+        {error, Reason} ->
+            error(Reason)
+    end;
+%% Example: data/user-credentials.csv
+reader(csv, Data, Convertor) when is_binary(Data) ->
```
Comment (zmstone) [inline apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl:505]: maybe add a `emqx_utils_stream:file_lines(Fd)` to create a stream.

File: apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl [hunk 2]
```diff
@@ -482,10 +445,143 @@ group_match_spec(UserGroup, QString) ->
...
+        end
+    end,
+    case get_csv_header(CSVData) of
+        {ok, CSVHeaders, CSVLines} ->
+            fun() -> Reader(CSVHeaders, CSVLines) end;
+        {error, Reason} ->
+            error(Reason)
+    end;
+%% Example: [#{<<"user_id">> => <<>>, ...}]
+reader(prepared_user_list, Data, Convertor) when is_list(Data) ->
```
Comment (zmstone) [inline apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl:532]: call `emqx_utils_stream:list/1` to create a stream.

File: changes/ce/feat-12396.en.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Enhanced the `authentication/:id/import_users` interface for a more user-friendly user import feature:
+
+- Add new parameter `?type=plain` to support importing users with plaintext passwords in extension to the current solution which only supports password hash.
```
Comment (ieQu1) [inline changes/ce/feat-12396.en.md:3]: Small nitpick:.
```suggestion
- Add new parameter `?type=plain` to support importing users with plaintext passwords in addition to the current solution which only supports password hash.
```

File: apps/emqx_utils/src/emqx_utils_stream.erl [hunk 4]
```diff
@@ -157,3 +164,48 @@ ets(Cont, ContF) ->
...
+
+csv_read_line(<<>>, Buff, Fields) ->
+    {lists:reverse([Buff | Fields]), <<>>};
+csv_read_line(<<"\r\n", Rest/binary>>, Buff, Fields) ->
+    {lists:reverse([Buff | Fields]), Rest};
+csv_read_line(<<"\n", Rest/binary>>, Buff, Fields) ->
+    {lists:reverse([Buff | Fields]), Rest};
+csv_read_line(<<"\r", Rest/binary>>, Buff, Fields) ->
+    {lists:reverse([Buff | Fields]), Rest};
+csv_read_line(<<", ", Rest/binary>>, Buff, Fields) ->
```
Comment (ieQu1) [inline apps/emqx_utils/src/emqx_utils_stream.erl:206]: Perhaps using something like `re:split` can simplify this function?

**Review by ieQu1 — APPROVED** [review]

File: apps/emqx_utils/src/emqx_utils_stream.erl [hunk 5]
```diff
@@ -157,3 +164,48 @@ ets(Cont, ContF) ->
                 []
         end
     end.
+
+%% @doc Make a stream out of a .csv binary, where the .csv binary is loaded in all at once.
+%% The .csv binary is assumed to be in UTF-8 encoding and to have a header row.
+-spec csv(binary()) -> stream(map()).
+csv(Bin) when is_binary(Bin) ->
```
Comment (zmstone) [inline apps/emqx_utils/src/emqx_utils_stream.erl:171]: why not base the implementation on `io:get_line` ?

**Review by zmstone — APPROVED** [review]

**Pull request merged**